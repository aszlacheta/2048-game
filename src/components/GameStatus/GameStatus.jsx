import { useTranslation } from "react-i18next";
import useGameStatus from "../../hooks/useGameStatus";
import "./GameStatus.scss";

/**
 * Component used to represent game status in the board view
 * @param {{hasMoreMoves: Boolean}} props React properties used to:
 *          1) hasMoreMoves - determines if user has more moves, true if there are possible moves; false otherwise 
 * @returns {ReactElement}
 */
export default function GameStatus({ hasMoreMoves, isWinner }) {
    const { t } = useTranslation();
    const status = useGameStatus(hasMoreMoves, isWinner);

    return (<div className="game-status">
        <span>{t("gameStatus.label")}</span>
        {status && <span data-status={status}>{t(`gameStatus.status.${status}`)}</span>}
    </div>)
}