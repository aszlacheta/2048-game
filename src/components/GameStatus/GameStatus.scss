@use '../variables.scss';

.game-status {
    margin-bottom: 2rem;

    color: gray;
}