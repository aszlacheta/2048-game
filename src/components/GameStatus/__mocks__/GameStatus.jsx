import React from 'react';

const GameStatus = () => <div className="mocked-game-status" />

module.exports = GameStatus;