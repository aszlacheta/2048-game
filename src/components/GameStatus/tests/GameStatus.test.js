import renderer from 'react-test-renderer';
import GameStatus from "../GameStatus";

describe('Game status component', () => {
    it('should render properly without props', () => {
        const component = renderer.create(<GameStatus />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('should render properly with hasMoreMoves provided', () => {
        const component = renderer.create(<GameStatus hasMoreMoves={true} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render properly with hasMoreMoves equals 0', () => {
        const component = renderer.create(<GameStatus hasMoreMoves={false} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render properly with wrong hasMoreMoves', () => {
        const component = renderer.create(<GameStatus hasMoreMoves={'a'} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});