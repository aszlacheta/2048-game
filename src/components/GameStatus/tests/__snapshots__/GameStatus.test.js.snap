// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game status component should render properly with hasMoreMoves equals 0 1`] = `
<div
  className="game-status"
>
  <span>
    gameStatus.label
  </span>
  <span
    data-status="game-over"
  >
    gameStatus.status.game-over
  </span>
</div>
`;

exports[`Game status component should render properly with hasMoreMoves provided 1`] = `
<div
  className="game-status"
>
  <span>
    gameStatus.label
  </span>
  <span
    data-status="playing"
  >
    gameStatus.status.playing
  </span>
</div>
`;

exports[`Game status component should render properly with wrong hasMoreMoves 1`] = `
<div
  className="game-status"
>
  <span>
    gameStatus.label
  </span>
  <span
    data-status="game-over"
  >
    gameStatus.status.game-over
  </span>
</div>
`;

exports[`Game status component should render properly without props 1`] = `
<div
  className="game-status"
>
  <span>
    gameStatus.label
  </span>
  <span
    data-status="game-over"
  >
    gameStatus.status.game-over
  </span>
</div>
`;
